### PR TITLE
Add world age and interpreter arguments to code_warntype.

### PR DESCRIPTION
**Pull request by maleadt**

### CODE REVIEW

**Review by simeonschaub — APPROVED** [review]
Seems reasonable to me.

**Review by vtjnash — APPROVED** [review]

File: stdlib/InteractiveUtils/src/codeview.jl [hunk 1]
```diff
@@ -57,10 +57,13 @@ Keyword argument `debuginfo` may be one of `:source` or `:none` (default), to sp
 
 See [`@code_warntype`](@ref man-code-warntype) for more information.
 """
-function code_warntype(io::IO, @nospecialize(f), @nospecialize(t); debuginfo::Symbol=:default, optimize::Bool=false)
+function code_warntype(io::IO, @nospecialize(f), @nospecialize(t);
+                       debuginfo::Symbol=:default, optimize::Bool=false,
```
Comment (vtjnash) [inline stdlib/InteractiveUtils/src/codeview.jl:61]: Perhaps we should have written this with kwargs forwarding?
```suggestion
                       debuginfo::Symbol=:default, kwargs...)
```

**Comment by oscardssmith** [comment]
Why does this need back-porting?

**Comment by maleadt** [comment]
> Why does this need back-porting?

It's a backport to the just-branched 1.7 release...

The GPU stack, and other packages, use their own AbstractInterpreter. Lacking a way to call reflection without these args, `code_warntype` is essentially broken for GPU code now that we start to rely more heavily on custom compilation (e.g. with https://github.com/JuliaLang/julia/pull/39697).

File: stdlib/InteractiveUtils/src/codeview.jl (hunk 1 above)
Comment (maleadt) [reply stdlib/InteractiveUtils/src/codeview.jl]: Yeah, that's better! I'll change it.

**Comment by maleadt** [comment]
The doctest failure is caused by this PR.

**Pull request merged**